Modules/Quotas/src/Quotas.Infrastructure/Persistence/Repository/IdentitiesRepository.cs

```diff
@@ -27,6 +27,15 @@ public async Task Add(Identity identity, CancellationToken cancellationToken)
         await _dbContext.SaveChangesAsync(cancellationToken);
     }
 
+    public async Task<Identity> FindByAddress(string address, CancellationToken cancellationToken, bool track = false)
+    {
+        var identity = await (track ? _identitiesDbSet : _readOnlyIdentities)
+            .IncludeAll(_dbContext)
+            .FirstWithAddress(address, cancellationToken);
```

Comment:
I thought about it, and we should probably rather return null instead of throwing a `NotFoundException`. Because this moves the logic of what to do when it was not found from the Infrastructure project into the Application project.
And yes, I know that we throw a NotFoundException from Infrastructure in a lot of other places as well. We will fix this some day. :)